src/app/components/AssignmentDutyForm.tsx

```diff
@@ -27,38 +36,134 @@ export interface AssignmentDutyFormProps {
     workSectionId?: WorkSectionCode;
 }
 
+interface SheriffDutyFieldProps {
+
+}
+
+class SheriffDutyFieldArray extends FieldArray<SheriffDutyFieldProps> {
+
+}
+
 export default class AssignmentDutyForm extends
     React.Component<AssignmentDutyFormProps & InjectedFormProps<{}, AssignmentDutyFormProps>, {}> {
+
+    static parseAssignmentDutyFromValues(values: any): AssignmentDuty {
+        const { timeRange: { startTime, endTime }, sheriffDuties, ...rest } = values;
+        const assignmentDuty = { ...rest };
+        assignmentDuty.startDateTime = startTime;
+        assignmentDuty.endDateTime = endTime;
+        assignmentDuty.sheriffDuties = sheriffDuties.map((element: any) => ({
+            startDateTime: moment(element.timeRange.startTime).toISOString(),
+            endDateTime: moment(element.timeRange.endTime).toISOString(),
+        }));
+        return assignmentDuty as AssignmentDuty;
+    }
+
+    static assignmentDutyToFormValues(duty: AssignmentDuty) {
+        return {
+            ...duty,
+            timeRange: {
+                startTime: moment(duty.startDateTime).toISOString(),
+                endTime: moment(duty.endDateTime).toISOString()
+            }
+        };
+    }
+
+    renderSheriffDutyFields(): React.ComponentClass {
```

Comment:
Since this returns us a component, perhaps we should change the name of it to `sheriffDutyFieldsComponent` to distinguish between something that's rendering actual components vs returning a component that needs to be instantiated.